examples/graphs/node2vec_movielens.py

```diff
@@ -0,0 +1,584 @@
+
+Our skip-gram is a simple binary classification model that works as follows:
+
+1. An embedding is looked up for the `target` movie.
+2. An embedding is looked up for the `context` movie.
+3. The dot product is computed between these two embeddings.
+4. The result (after a sigmoid activation) is compared to the label.
+5. A binary crossentropy loss is used.
+"""
+
+learning_rate = 0.001
+embedding_dim = 50
+num_epochs = 10
+
+"""
+### Implement the model
+"""
+
+
+def create_model(vocabulary_size, embedding_dim):
+
+    inputs = {
+        "target": layers.Input(name="target", shape=(), dtype="int32"),
+        "context": layers.Input(name="context", shape=(), dtype="int32"),
+    }
+    # Initialize item embeddings.
+    embed_item = layers.Embedding(
+        input_dim=vocabulary_size,
+        output_dim=embedding_dim,
+        embeddings_initializer="he_normal",
+        embeddings_regularizer=keras.regularizers.l2(1e-6),
+        name="item_embeddings",
+    )
+    # Lookup embeddings for target.
+    target_embeddings = embed_item(inputs["target"])
+    # Lookup embeddings for context.
+    context_embeddings = embed_item(inputs["context"])
+    # Compute dot similarity between target and context embeddings.
+    logits = layers.Dot(axes=1, normalize=False, name="dot_similarity")(
+        [target_embeddings, context_embeddings]
+    )
+    # Create the model.
+    model = keras.Model(inputs=inputs, outputs=logits)
+    return model
+
+
+"""
+### Train the model
+"""
+
+"""
+We instantiate the model and compile it.
+"""
+
+model = create_model(len(vocabulary), embedding_dim)
+model.compile(
+    optimizer=keras.optimizers.Adam(learning_rate),
+    loss=keras.losses.BinaryCrossentropy(from_logits=True),
+)
+
+"""
+Let's plot the model.
+"""
+
+keras.utils.plot_model(
+    model, show_shapes=True, show_dtype=True, show_layer_names=True,
+)
+
+"""
+Now we fit the model with the `dataset`.
+"""
+
+history = model.fit(dataset, epochs=num_epochs)
+
+"""
+Finally we plot the learning history.
+"""
+
+plt.plot(history.history["loss"])
+plt.ylabel("loss")
+plt.xlabel("epoch")
+plt.show()
+
+"""
+## Analyze the learnt embeddings
+"""
+
+movie_embeddings = model.get_layer("item_embeddings").get_weights()[0]
+print("Embeddings shape:", movie_embeddings.shape)
+
+"""
+### Find related movies
+"""
+
+query_movies = [
+    "Matrix, The (1999)",
+    "Star Wars: Episode IV - A New Hope (1977)",
+    "Lion King, The (1994)",
+    "Terminator 2: Judgment Day (1991)",
+    "Godfather, The (1972)",
+]
+
+"""
+Get the embeddings of the query movies
+"""
+
+query_embeddings = []
+
+for movie_title in query_movies:
+    movieId = get_movie_id_by_title(movie_title)
+    token_id = vocabulary_lookup[movieId]
+    movie_embedding = movie_embeddings[token_id]
+    query_embeddings.append(movie_embedding)
+
+query_embeddings = np.array(query_embeddings)
+
+"""
+Compute the consine similarity between the embeddings of the query movies and all
```

Comment:
Suggestion:

- Add a link to https://en.wikipedia.org/wiki/Cosine_similarity to help new users

```suggestion
Compute the [consine similarity](https://en.wikipedia.org/wiki/Cosine_similarity) between the embeddings of the query movies and all
```